File: Python/8kyu/8kyu-counting-sheep.py
# Counting sheep...
# Language: Python
# Version: 3.10
# Difficulty: 8 kyu

def count_sheeps(sheep):
  return sheep.count(True)


# Tests
import codewars_test as test
from solution import count_sheeps

@test.describe("Basic Tests")
def basic_tests():
    
    @test.it("Basic Tests")
    def basic_tests():
        array1 = [True,  True,  True,  False,
                  True,  True,  True,  True ,
                  True,  False, True,  False,
                  True,  False, False, True ,
                  True,  True,  True,  True ,
                  False, False, True,  True ];
        
        result = count_sheeps(array1)
        test.assert_equals(result, 17, f"There are 17 sheeps in total, not {result}")

File: Julia/8kyu/8kyu-counting-sheep.jl
# Counting sheep...
# Language: Julia
# Version: 1.5
# Difficulty: 8 kyus

# Solution 1
function countsheep(sheep) 
    count(sheep)
end

# Solution 2
function countsheep(sheep) 
    sum(sheep)
end


# Tests
using FactCheck

facts("Basic tests") do
  context("Should pass fixed tests") do
    array1 = [true, true, true, false, true, true, true, true, true, false, true, false, true, false, false, 
              true, true, true, true, true, false, false, true, true ]
    @fact countsheep(array1) --> 17
  end
end